src/specfem3D/setup_sources_receivers.f90

```diff
@@ -123,6 +123,9 @@ subroutine setup_sources()
     allocate(pm1_source_encoding(1),stat=ier)
   endif
   if (ier /= 0) stop 'error allocating arrays for sources'
+  
+  allocate(user_source_time_function(NSTEP, NSOURCES),stat=ier)
```

Comment:
Hi Vadim,

I agree with Daniel. Could you please fix that tomorrow and submit a 
pull request?
(I have already accepted https://github.com/geodynamics/specfem3d/pull/762 )

thanks
Dimitri

On 01/06/2016 21:00, daniel peter wrote:
> In src/specfem3D/setup_sources_receivers.f90
> <https://github.com/geodynamics/specfem3d/pull/762#discussion_r65422033>:
>
>> @@ -123,6 +123,9 @@ subroutine setup_sources()
>>      allocate(pm1_source_encoding(1),stat=ier)
>>    endif
>>    if (ier /= 0) stop 'error allocating arrays for sources'
>> +
>> +  allocate(user_source_time_function(NSTEP, NSOURCES),stat=ier)
>
> hi Vadim,
> better if this would only get allocated in case EXTERNAL_STF is actually
> used. the array can get very big for cases with many sources (kinematic
> rupture simulations) and/or longer simulation times.
>
> if EXTERNAL_STF is not used, either allocate a dummy array
> (user_source_time_function(1,1)) or don't allocate at all and remove if
> from subroutine argument calls (put it in the "use specfem_par,only: .."
> part).
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem3d/pull/762/files/fcc507bdb1c432d2b934bde0656b6c123925a123#r65422033>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFjDKQdGLfzChxxomMAFe8YUZR9uCeXaks5qHdazgaJpZM4IrueX>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr